server/channels/app/plugin_requests.go

```diff
@@ -158,96 +188,73 @@ func (ch *Channels) servePluginRequest(w http.ResponseWriter, r *http.Request, h
+			sentToken = r.Header.Get(model.HeaderCsrfToken)
+		}
 
-			expectedToken := session.GetCSRF()
-
-			if sentToken == expectedToken {
-				csrfCheckPassed = true
-			}
+		expectedToken := session.GetCSRF()
 
-			// ToDo(DSchalla) 2019/01/04: Remove after deprecation period and only allow CSRF Header (MM-13657)
-			if r.Header.Get(model.HeaderRequestedWith) == model.HeaderRequestedWithXML && !csrfCheckPassed {
-				csrfErrorMessage := "CSRF Check failed for request - Please migrate your plugin to either send a CSRF Header or Form Field, XMLHttpRequest is deprecated"
-				sid := ""
-				userID := ""
-
-				if session.Id != "" {
-					sid = session.Id
-					userID = session.UserId
-				}
-
-				fields := []mlog.Field{
-					mlog.String("path", r.URL.Path),
-					mlog.String("ip", r.RemoteAddr),
-					mlog.String("session_id", sid),
-					mlog.String("user_id", userID),
-				}
-
-				if *ch.cfgSvc.Config().ServiceSettings.ExperimentalStrictCSRFEnforcement {
-					mlog.Warn(csrfErrorMessage, fields...)
-				} else {
-					mlog.Debug(csrfErrorMessage, fields...)
-					csrfCheckPassed = true
-				}
-			}
-		} else {
+		if sentToken == expectedToken {
 			csrfCheckPassed = true
 		}
 
-		if (session != nil && session.Id != "") && appErr == nil && csrfCheckPassed {
-			r.Header.Set("Mattermost-User-Id", session.UserId)
-			context.SessionId = session.Id
+		// ToDo(DSchalla) 2019/01/04: Remove after deprecation period and only allow CSRF Header (MM-13657)
+		if r.Header.Get(model.HeaderRequestedWith) == model.HeaderRequestedWithXML && !csrfCheckPassed {
+			csrfErrorMessage := "CSRF Check failed for request - Please migrate your plugin to either send a CSRF Header or Form Field, XMLHttpRequest is deprecated"
+			sid := ""
+			userID := ""
 
-			r.Header.Del(model.HeaderAuth)
-		}
-	}
+			if session.Id != "" {
+				sid = session.Id
+				userID = session.UserId
+			}
 
-	cookies := r.Cookies()
-	r.Header.Del("Cookie")
-	for _, c := range cookies {
-		if c.Name != model.SessionCookieToken {
-			r.AddCookie(c)
+			fields := []mlog.Field{
+				mlog.String("path", r.URL.Path),
+				mlog.String("ip", r.RemoteAddr),
+				mlog.String("session_id", sid),
+				mlog.String("user_id", userID),
+			}
+
+			if *ch.cfgSvc.Config().ServiceSettings.ExperimentalStrictCSRFEnforcement {
+				mlog.Warn(csrfErrorMessage, fields...)
+			} else {
+				mlog.Debug(csrfErrorMessage, fields...)
+				csrfCheckPassed = true
+			}
 		}
+	} else {
+		csrfCheckPassed = true
 	}
-	r.Header.Del("Referer")
 
-	params := mux.Vars(r)
+	if (session != nil && session.Id != "") && appErr == nil && csrfCheckPassed {
+		r.Header.Set("Mattermost-User-Id", session.UserId)
+		context.SessionId = session.Id
 
-	subpath, _ := utils.GetSubpathFromConfig(ch.cfgSvc.Config())
-
-	newQuery := r.URL.Query()
-	newQuery.Del("access_token")
-	r.URL.RawQuery = newQuery.Encode()
-	r.URL.Path = strings.TrimPrefix(r.URL.Path, path.Join(subpath, "plugins", params["plugin_id"]))
+		r.Header.Del(model.HeaderAuth)
```

Comment:
Should we be deleting this upfront, and unconditionally? Seems weird to only delete it for CSRF failing, and not for MFA missing.

Reply:
Good point, see https://github.com/mattermost/mattermost/pull/30795/commits/3141688ff2d6ebc4c64d2e33096a79ba7f26db36

Reply:
Looks like I was wrong to suggest doing this unconditionally, as the `Authorization` header can be used for auth coordinated with third-party systems.